ophyd/mixins.py

```diff
@@ -0,0 +1,151 @@
+    def egu(self):
+        '''The engineering units (EGU) for positions'''
+        return self._egu
+
+    def move(self, position, wait=True, moved_cb=None, timeout=None):
+        '''Move to a specified position, optionally waiting for motion to
+        complete.
+
+        Parameters
+        ----------
+        position
+            Position to move to
+        wait : bool, optional
+            Wait until motion has completed before returning
+        moved_cb : callable
+            Call this callback when movement has finished. This callback
+            must accept one keyword argument: 'obj' which will be set to
+            this positioner instance.
+        timeout : float, optional
+            Maximum time to wait for the motion. If None, the default timeout
+            for this positioner is used.
+
+        Returns
+        -------
+        status : Status
+
+        Raises
+        ------
+        TimeoutError
+            When motion takes longer than `timeout`
+        ValueError
+            On invalid positions
+        RuntimeError
+            If motion fails other than timing out
+        '''
+        if timeout is None:
+            timeout = self._timeout
+
+        self.check_value(position)
+
+        self._run_subs(sub_type=self._SUB_REQ_DONE, success=False)
+        self._reset_sub(self._SUB_REQ_DONE)
+
+        self._moving = True
+        self._run_subs(sub_type=self.SUB_START)
+
+        def finished():
+            success = self._internal_status.success
+            self._done_moving(success=success)
+
+            if moved_cb is not None:
+                try:
+                    moved_cb(obj=self)
+                except Exception as ex:
+                    logger.error('Move callback failed', exc_info=ex)
+
+            try:
+                self._external_status._finished(success=success)
+            except Exception as ex:
+                logger.error('Status completion failed', exc_info=ex)
+
+            self._internal_status = None
+            self._external_status = None
+
+        # this external status object is fully dependent on the internal status
+        # object and does not have its own timeout/settle_time settings:
+        self._external_status = MoveStatus(self, target=position,
+                                           settle_time=0.0, timeout=None)
+
+        # set() functionality depends on the signal
+        self._internal_status = self._mixed_set(position, timeout=timeout,
+                                                settle_time=self.settle_time)
+        self._internal_status.finished_cb = finished
+
+        if wait:
+            try:
+                status_wait(self._internal_status)
+            except RuntimeError:
+                raise RuntimeError('Motion did not complete successfully')
+
+        return self._external_status
+
+    def stop(self):
+        '''Stops motion'''
+        if self._hold_on_stop:
+            self.move(self.get(), wait=False)
```

Comment:
If there's nothing but a setpoint and a readback, what else are you supposed to do?